### ISSUE TITLE
[bug] @eslint-react/react/no-dangerously-set-innerhtml-with-children doesn't fail with failing example

### ISSUE THREAD

**Issue by karlhorky** [Rel1cx/eslint-react#1163]
### Describe the bug

Hi @Rel1cx, hope you're well! 👋

I noticed that the failing example code for [`@eslint-react/react/no-dangerously-set-innerhtml-with-children`](https://eslint-react.xyz/docs/rules/dom-no-dangerously-set-innerhtml-with-children) doesn't fail:

<img width="2544" height="810" alt="Image" src="https://github.com/user-attachments/assets/4f85dc51-3cd9-4a1d-8d01-07d9d7a4651e" />

The div with the text node as children errors (similar to [the failing example code from `eslint-plugin-react`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-danger-with-children.md)), but the div with the JSX element as children does not

```bash
$ pnpm eslint .

/Users/k/p/repro-eslint-react-dom-no-dangerously-set-innerhtml-with-children-with-typescript-eslint/index.tsx
  1:10  error  'Abc' is defined but never used                                           @typescript-eslint/no-unused-vars
  5:7   error  A DOM component cannot use both 'children' and 'dangerouslySetInnerHTML'  @eslint-react/dom/no-dangerously-set-innerhtml-with-children

✖ 2 problems (2 errors, 0 warnings)
```

### Reproduction

- https://github.com/karlhorky/repro-eslint-react-dom-no-dangerously-set-innerhtml-with-children-with-typescript-eslint

### Expected behavior

The failing example code for [`@eslint-react/react/no-dangerously-set-innerhtml-with-children`](https://eslint-react.xyz/docs/rules/dom-no-dangerously-set-innerhtml-with-children) (div with the JSX element as children) should error

```tsx
import React from "react";

function MyComponent() {
  return (
    <div dangerouslySetInnerHTML={{ __html: "Hello World" }}>
      <p>Goodbye World</p>
    </div>
  );
}
```

### Platform and versions

```bash
$ node -v
v22.18.0

$ cat package.json
...
  "dependencies": {
    "react": "19.1.1"
  },
  "devDependencies": {
    "@eslint-react/eslint-plugin": "1.52.3",
    "@eslint/js": "9.32.0",
    "@types/react": "19.1.9",
    "eslint": "9.32.0",
    "typescript": "5.8.3",
    "typescript-eslint": "8.38.0"
  },
...

$ cat eslint.config.mjs
import eslintReact from '@eslint-react/eslint-plugin';
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommended,
  eslintReact.configs.recommended,
  eslintReact.configs['disable-dom'],
  {
    rules: {
      '@eslint-react/dom/no-dangerously-set-innerhtml-with-children': 'error',
    },
  },
);
```

### Stack trace

```Text

```

**Comment by karlhorky** [comment]
@Rel1cx thanks for the fix in PR #1165 and the release!

I can confirm that the new release [`eslint-plugin-react-dom@2.0.0-beta.23`](https://www.npmjs.com/package/eslint-plugin-react-dom/v/2.0.0-beta.23) resolved the issue:

<img width="1902" height="630" alt="Image" src="https://github.com/user-attachments/assets/3695300a-0f62-40c7-b30c-44759960c292" />
